### PR TITLE
[AN-4490] Added emoji picker for sketch

### PR DESCRIPTION
**Pull request by hakonbo**
Not in this PR:
- Emoji group headers

#### APK
[Download build #7578](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7578/artifact/build/artifact/wire-dev-PR150-7578.apk)
[Download build #7579](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7579/artifact/build/artifact/wire-dev-PR150-7579.apk)
[Download build #7581](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7581/artifact/build/artifact/wire-dev-PR150-7581.apk)
[Download build #7582](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7582/artifact/build/artifact/wire-dev-PR150-7582.apk)
[Download build #7584](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7584/artifact/build/artifact/wire-dev-PR150-7584.apk)

### CODE REVIEW

**Comment by hakonbo** [comment]
build again

**Comment by xuanzhaopeng** [comment]
Test it on Phone and Tablet, except ONLY on [Tablet] Samsung SM-T310, the emoji is too small, in other devices works fine.   Blank emojis fixed.

File: app/src/main/java/com/waz/zclient/controllers/userpreferences/RecentEmojis.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import org.json.JSONException;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class RecentEmojis {
+
+    private static final int MAX_RECENT_EMOJIS = 12;
+
+    private List<String> recentEmojis;
```
Comment (deanrobertcook) [inline app/src/main/java/com/waz/zclient/controllers/userpreferences/RecentEmojis.java]: Apparently there's a thing called [LinkedHashSet](http://stackoverflow.com/a/2319126/1751834) which might be better here?

File: wire-core/src/main/java/com/waz/zclient/utils/Emojis.java [hunk 2]
```diff
@@ -0,0 +1,1321 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.waz.zclient.utils;
+
+/**
+ * There seems to be an issue with decoding XML with emojis on some devices
+ * https://code.google.com/p/android/issues/detail?id=81341
```
Comment (deanrobertcook) [inline wire-core/src/main/java/com/waz/zclient/utils/Emojis.java]: Might need to provide more attribution?

File: wire-core/src/main/res/values/emojis.xml [hunk 3]
```diff
@@ -0,0 +1,1316 @@
...
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+-->
+<resources>
+
+    <string-array name="emoji_flags">
```
Comment (deanrobertcook) [inline wire-core/src/main/res/values/emojis.xml]: Attribution

File: wire-ui/src/main/java/com/waz/zclient/ui/colorpicker/EmojiBottomSheetDialog.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    private void populateEmojis(List<String> recent, Set<String> unsupported) {
+        spaces = new ArrayList<>();
+        emojis = new ArrayList<>();
+        if (recent != null && recent.size() > 0) {
+            emojis.addAll(recent);
+            spaces.add(emojis.size());
+            emojis.add(EmojiAdapter.SPACE);
+        }
+        emojis.addAll(getFilteredList(Emojis.PEOPLE, unsupported));
+        spaces.add(emojis.size());
```
Comment (deanrobertcook) [inline wire-ui/src/main/java/com/waz/zclient/ui/colorpicker/EmojiBottomSheetDialog.java]: Might be nicer to keep a set of the different emoji sets in the `Emojis` class, and then you could iterate through that set for logic like this

**Comment by deanrobertcook** [comment]
Review done

**Comment by hakonbo** [comment]
build again

**Pull request merged**